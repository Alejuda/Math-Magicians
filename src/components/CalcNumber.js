import PropTypes from 'prop-types';

const CalcNumber = ({ number }) => (
  <div className={`calc-number calc-${number}`}>{number}</div>
);

export default CalcNumber;

CalcNumber.propTypes = {
  number: PropTypes.number.isRequired,
};
