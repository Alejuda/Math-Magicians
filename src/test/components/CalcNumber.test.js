import { render, fireEvent } from "@testing-library/react";
import CalcNumber from "../../components/CalcNumber";

describe("CalcNumber component", () => {
  const onClickMock = jest.fn(console.log("clicked"));

  it("should renders correctly", () => {
    const { container } = render(
      <CalcNumber number={2} handleClick={onClickMock} />
    );
    expect(container).toMatchSnapshot();
  });

  it("click event calls the provided function", () => {
    const { getByRole } = render(
      <CalcNumber number={2} handleClick={onClickMock} />
    );

    const element = getByRole("button");
    fireEvent.click(element);

    expect(onClickMock).toHaveBeenCalledTimes(1);
  });
});
