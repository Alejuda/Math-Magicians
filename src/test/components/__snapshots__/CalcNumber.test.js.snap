// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalcNumber component should renders correctly 1`] = `
<div>
  <div
    class="calc-number calc-2"
    role="button"
    tabindex="0"
  >
    2
  </div>
</div>
`;
